ion-schema/src/isl/version_based_isl/constraints/any_of.rs

```diff
@@ -0,0 +1,16 @@
+use crate::isl::version_based_isl::{IslTypeArgument, IslVersionTrait};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct AnyOf<V: IslVersionTrait> {
+    type_args: Vec<IslTypeArgument<V>>,
+}
+
+impl<V: IslVersionTrait> AnyOf<V> {
+    pub fn new(type_args: Vec<IslTypeArgument<V>>) -> Self {
+        Self { type_args }
+    }
+
+    pub fn type_args(&self) -> &Vec<IslTypeArgument<V>> {
```

Comment:
For all of these of accessor methods, return a slice (`&[IslTypeArgument]`) instead of a reference to the `Vec` to avoid double indirection.